### PR TITLE
samples: openthread: enable rebooting to bootloader for nRF52 Dongle

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
enable rebooting to bootloader for nRF52 Dongle with `ot reset bootloader` command triggering GPIO pin.

### CODE REVIEW

**Review by edmont — APPROVED** [review]

File: samples/openthread/coprocessor/README.rst [hunk 1]
```diff
@@ -78,6 +78,13 @@ Diagnostic module
 The Co-processor sample enables a diagnostic module in a similar way as described in the :ref:`ot_cli_sample_diag_module` section of the :ref:`ot_cli_sample` sample documentation.
 However, the Co-processor and CLI samples use different commands for the module, as described in the :ref:`ot_coprocessor_testing` section.
 
+Rebooting to bootloader
+=======================
+
+The Co-processor sample enables rebooting to bootloader for the ``nrf52840dongle_nrf52840`` build target, in a similar way as described in the :ref:`ot_cli_sample_bootloader` section of the :ref:`ot_cli_sample` sample documentation.
```
Comment (wiba-nordic) [inline samples/openthread/coprocessor/README.rst:84]:
```suggestion
The Co-processor sample enables rebooting to bootloader for the ``nrf52840dongle_nrf52840`` build target, similar to what is described in the :ref:`ot_cli_sample_bootloader` section of the :ref:`ot_cli_sample` sample documentation.
```

File: samples/openthread/coprocessor/README.rst [hunk 2]
```diff
@@ -78,6 +78,13 @@ Diagnostic module
...
+However, the Co-processor and CLI samples use different commands, as described in the :ref:`ot_coprocessor_testing` section.
+Additionally, posix application described in :ref:`ug_thread_tools_ot_apps` should be built with ``-DOT_PLATFORM_BOOTLOADER_MODE=ON`` option.
```
Comment (wiba-nordic) [inline samples/openthread/coprocessor/README.rst:86]:
```suggestion
Additionally, the :ref:`ug_thread_tools_ot_apps` should be built with ``-DOT_PLATFORM_BOOTLOADER_MODE=ON`` option.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: samples/openthread/cli/README.rst [hunk 3]
```diff
@@ -64,20 +64,21 @@ See `Testing diagnostic module`_ section for an example.
...
-For the ``nrf52840dongle_nrf52840`` build target, diagnostic GPIO commands can also be used.
-They are enabled in the ``openthread_config`` node in the :file:`boards/nrf52840dongle_nrf52840.overlay` file, with the node configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle.
-The pin is set in output mode with a low state so that the **RESET** pin is pulled to **GND**, which results in the device rebooting without skipping the bootloader.
+.. _ot_cli_sample_bootloader:
+
+Rebooting to bootloader
+=======================
+
+For the ``nrf52840dongle_nrf52840`` build target, rebooting to bootloader by triggering GPIO pin is possible after enabling :kconfig:option:`CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_GPIO` and configuring Devicetree overlay.
+In the ``openthread_config`` node in the :file:`boards/nrf52840dongle_nrf52840.overlay` file, property ``bootloader-gpios`` is configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle.
```
[the remark this answers is not in the document]
Comment (maciejbaczmanski) [reply samples/openthread/cli/README.rst:73]: I'd use "is configured" as it is already done in the sample under `boards/nrf52840dongle_nrf52840.overlay`

[the remark this answers is not in the document]
Comment (maciejbaczmanski) [reply samples/openthread/cli/README.rst:73]: @wiba-nordic  maybe we could ad "For this sample" at the beginning? :
"For this sample, the ``openthread_config`` node in the ``bootloader-gpios`` property is configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle."

[the remark this answers is not in the document]
Comment (maciejbaczmanski) [reply samples/openthread/cli/README.rst]: also, it should be "property in the node is configured" not the other way

File: samples/openthread/cli/README.rst [hunk 4]
```diff
@@ -64,20 +64,21 @@ See `Testing diagnostic module`_ section for an example.
...
-They are enabled in the ``openthread_config`` node in the :file:`boards/nrf52840dongle_nrf52840.overlay` file, with the node configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle.
-The pin is set in output mode with a low state so that the **RESET** pin is pulled to **GND**, which results in the device rebooting without skipping the bootloader.
+.. _ot_cli_sample_bootloader:
+
+Rebooting to bootloader
+=======================
+
+For the ``nrf52840dongle_nrf52840`` build target, the device can reboot to bootloader by triggering a GPIO pin.
+To enable this behavior, enable the :kconfig:option:`CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_GPIO` Kconfig option and configure the Devicetree overlay in the :file:`boards/nrf52840dongle_nrf52840.overlay` file.
+For this sample the ``bootloader-gpios`` property in the ``openthread_config`` node is configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle.
```
Comment (wiba-nordic) [inline samples/openthread/cli/README.rst:74]:
```suggestion
For this sample, the ``bootloader-gpios`` property in the ``openthread_config`` node is pre-configured for the **P0.19** pin, which is connected to the **RESET** pin on the nRF52840 Dongle.
```
I think changing it to "pre-configured" might be good, so it doesn't seem at odds with telling the user to configure the Devicetree overlay.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: samples/openthread/cli/README.rst (hunk 4 above)
Comment (maciejbaczmanski) [reply samples/openthread/cli/README.rst]: Makes sense, fixed.

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**